hack/reduce-snapshot/reduce-snapshot.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+set -eux
+
+echo "Single Component mode? ${SINGLE_COMPONENT}"
+if [ "${SINGLE_COMPONENT}" == "true" ]; then
+
+  CR_NAMESPACE_ARG=
+  if [ "${CUSTOM_RESOURCE_NAMESPACE}" != ""]; then
+    CR_NAMESPACE_ARG="-n ${CUSTOM_RESOURCE_NAMESPACE}"
+  fi
+  SNAPSHOT_CREATION_TYPE=$(kubectl get "$CUSTOM_RESOURCE" ${CR_NAMESPACE_ARG:+$CR_NAMESPACE_ARG} -ojson \
+      | jq -r '.metadata.labels."test.appstudio.openshift.io/type" // ""')
+  SNAPSHOT_CREATION_COMPONENT=$(kubectl get "$CUSTOM_RESOURCE" ${CR_NAMESPACE_ARG:+$CR_NAMESPACE_ARG} -ojson \
+      | jq -r '.metadata.labels."appstudio.openshift.io/component" // ""')
+
+  echo "SNAPSHOT_CREATION_TYPE: ${SNAPSHOT_CREATION_TYPE}"
+  echo "SNAPSHOT_CREATION_COMPONENT: ${SNAPSHOT_CREATION_COMPONENT}"
+  if [ "${SNAPSHOT_CREATION_TYPE}" == "component" ] && [ "${SNAPSHOT_CREATION_COMPONENT}" != "" ]; then
+    echo "Single Component mode is ${SINGLE_COMPONENT} and Snapshot type is component"
+    # verify if in json form
+    set +e
+    echo "${SNAPSHOT}" | jq -e > /dev/null 2>&1
+    CHECK_JSON=$?
+    set -e
+    if [ "${CHECK_JSON}" != "0" ]; then
+      # try to load as file first
+      set +e
+      jq -e < "${SNAPSHOT}" > /dev/null 2>&1
+      CHECK_JSON_FILE=$?
+      set -e
+      if [ "${CHECK_JSON_FILE}" != "0" ]; then
+        echo "Cannot load snapshot from json string or file"
+        exit 1
+      fi
+      SNAPSHOT=$(cat "${SNAPSHOT}")
+    fi
```

Comment:
I think the `jq` commands here lack the filter. I think something simpler can be done here instead:

```suggestion
    SNAPSHOT="$(cat "${SNAPSHOT}" 2> /dev/null || echo "${SNAPSHOT}")"
    [[ ! "${SNAPSHOT}" =~ ^\s*\{ ]] && echo 'Error: Cannot load snapshot from JSON string or file' && exit 1
```